src/main/java/gov/usgs/wma/waterdata/format/WaterMLPointToXmlResultHandler.java

```diff
@@ -34,33 +34,41 @@
     private String monLocIdentifier = "";
     private String pcode = "";
 
+    // number of results processed
+    private int numResults = 0;
+
     // elements in xml still pending
-    private Stack<String> openElements = new Stack<>();
+    private final Stack<String> openElements = new Stack<>();
 
     public WaterMLPointToXmlResultHandler(Writer writer) throws XMLStreamException {
         this.writer = XmlFactory.newXMLStreamWriter(writer);
-        initDoc();
     }
 
     public WaterMLPointToXmlResultHandler(OutputStream out) throws XMLStreamException {
         this.writer = XmlFactory.newXMLStreamWriter(out);
-        initDoc();
     }
 
     @Override
     public void handleResult(final ResultContext<? extends WaterMLPoint> resultContext) {
         WaterMLPoint point = resultContext.getResultObject();
-        boolean s = false;
         try {
+            if(numResults == 0) {
+                initDoc();
+            }
```

Comment:
it feels like this init belongs someplace else.

Reply:
It was here because we can not write to the response output stream until we know there was  data found. 
So the output would just contain "requested data not found" and not include the WaterML header.